Update html.css to use logical properties for form controls

pxlcoder opened the pull request:
#### 55af19e16ebb473a7a4f4d5a4fe07c23179c5259
<pre>
Update html.css to use logical properties for form controls
<a href="https://bugs.webkit.org/show_bug.cgi?id=264236">https://bugs.webkit.org/show_bug.cgi?id=264236</a>
<a href="https://rdar.apple.com/117983666">rdar://117983666</a>

Reviewed by Tim Nguyen.

This is necessary in order to support vertical form controls.

* Source/WebCore/css/html.css:
(input):
(input[type=&quot;search&quot;]::-webkit-search-cancel-button,input[type=&quot;search&quot;]::-webkit-search-decoration,input[type=&quot;search&quot;]::-webkit-search-results-decoration):
(input[type=&quot;search&quot;]::-webkit-search-results-button):
(input::-webkit-date-and-time-value:
(input::-webkit-list-button):
(textarea):
(select):

Canonical link: <a href="https://commits.webkit.org/270277@main">https://commits.webkit.org/270277@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/78261c5891ed2e2e4e998ea580e0caa71360b2b5

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/25043 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/3583 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/26296 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/27159 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/22989 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/25311 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/5286 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/1023 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/23267 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/25287 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/2620 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/21620 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/27739 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/2325 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/22553 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/28685 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/22850 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/22909 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/26502 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/2258 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/560 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/3540 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/6001 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/2703 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/2601 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/css/html.css:
```diff
@@ -769,8 +774,8 @@ textarea {
-    padding-top: 2px;
-    padding-bottom: 2px;
+    padding-block-start: 2px;
+    padding-block-end: 2px;
```
```suggestion
    padding-block: 2px;
```

pxlcoder replied:
Done.

nt1m reviewed (approved):
I originally left these unchanged to be able to use html.css as a checklist, but the code change is straightforward and not user facing, so this is fine.

nt1m commented on Source/WebCore/css/html.css:
```diff
@@ -738,13 +743,13 @@ input::-webkit-list-button {
-    width: 11px;
+    inline-size: 11px;
     /* Make it easier to hit the button on iOS */
     padding: 7px;
     margin: -7px;
 #else
-    width: 16px;
-    height: 100%;
+    inline-size: 16px;
+    block-size: 100%;
```
I'm actually unsure if this is correct without some platform changes, since this button is drawn by the platform. If we're not rotating the datalist button, then this might not be fully correct!

pxlcoder replied:
I have another patch in progress to fix the painting of the button, but this is correct in terms of positioning the element.

pxlcoder committed:
Update html.css to use logical properties for form controls
https://bugs.webkit.org/show_bug.cgi?id=264236
rdar://117983666

Reviewed by Tim Nguyen.

This is necessary in order to support vertical form controls.

* Source/WebCore/css/html.css:
(input):
(input[type="search"]::-webkit-search-cancel-button,input[type="search"]::-webkit-search-decoration,input[type="search"]::-webkit-search-results-decoration):
(input[type="search"]::-webkit-search-results-button):
(input::-webkit-date-and-time-value:
(input::-webkit-list-button):
(textarea):
(select):

Canonical link: https://commits.webkit.org/270277@main
